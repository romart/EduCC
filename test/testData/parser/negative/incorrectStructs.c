struct SS {
    int a;
    int b, d;
    typedef struct {
        double x;
        struct XXX ****xx;
    } ss1, *ss2, **ss3;

    ss1 v1, *pv1, **pv1;
    ss2 v2, *pv2, **pv2;
    ss3 v3, *pv3, **pv3;
};


