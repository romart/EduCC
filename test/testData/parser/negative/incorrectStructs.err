test/testData/parser/negative/incorrectStructs.c:5:4: error: type name does not allow storage class to be specified
test/testData/parser/negative/incorrectStructs.c:9:9: error: unexpected token IDENTIFIER 'v1' instead of ;
test/testData/parser/negative/incorrectStructs.c:9:10: error: unexpected token IDENTIFIER 'v2' instead of ;
test/testData/parser/negative/incorrectStructs.c:9:11: error: unexpected token IDENTIFIER 'v3' instead of ;
