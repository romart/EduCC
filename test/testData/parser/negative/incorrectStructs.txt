FILE test/testData/parser/negative/incorrectStructs.c
  STRUCT SS
    signed int a #0
    signed int b #4
    signed int d #8
    STRUCT <anon$0>
      double x #0
      ****STRUCT XXX xx #8
    STRUCT_END
    STRUCT <anon$0> ss1 #16
    *STRUCT <anon$0> ss2 #32
    **STRUCT <anon$0> ss3 #40
    ERROR TYPE ss1 #0
    ERROR TYPE v1 #1
    *ERROR TYPE pv1 #0
    **ERROR TYPE pv1 #8
    ERROR TYPE ss2 #0
    ERROR TYPE v2 #1
    *ERROR TYPE pv2 #0
    **ERROR TYPE pv2 #8
    ERROR TYPE ss3 #0
    ERROR TYPE v3 #1
    *ERROR TYPE pv3 #0
    **ERROR TYPE pv3 #8
  STRUCT_END
