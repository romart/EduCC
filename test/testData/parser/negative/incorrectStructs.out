test/testData/parser/negative/incorrectStructs.c:5:9: warning: type specifier missing, defaults to 'int'
test/testData/parser/negative/incorrectStructs.c:11:9: warning: type specifier missing, defaults to 'int'
test/testData/parser/negative/incorrectStructs.c:11:9: warning: declaration does not declare anything
test/testData/parser/negative/incorrectStructs.c:5:10: warning: type specifier missing, defaults to 'int'
test/testData/parser/negative/incorrectStructs.c:11:10: warning: type specifier missing, defaults to 'int'
test/testData/parser/negative/incorrectStructs.c:11:10: warning: declaration does not declare anything
test/testData/parser/negative/incorrectStructs.c:5:11: warning: type specifier missing, defaults to 'int'
test/testData/parser/negative/incorrectStructs.c:11:11: warning: type specifier missing, defaults to 'int'
test/testData/parser/negative/incorrectStructs.c:11:11: warning: declaration does not declare anything
